docs/upgrade/11_to_12/workflows_postgresql.py

```diff
@@ -0,0 +1,427 @@
+      operations.append(0)
+    elif get_attrib_from_node(operation, "retry-strategy") == "retry":
+      operations.append(1)
+    elif get_attrib_from_node(operation, "retry-strategy") == "hold":
+      operations.append(2)
+    else:
+      operations.append(None)
+    operations.append(operation_position)
+    operations.append(workflow_id)
+    operations.append(operation_position)
+
+    wf_operation.append(operations)
+
+    operation_position += 1
+
+    ### oc_workflow_operation_configuration
+    for op_config in operation.find("{http://workflow.opencastproject.org}configurations"):
+      op_configs = []
+
+      op_configs.append(operation_id)
+      op_configs.append(get_attrib_from_node(op_config, "key"))
+      op_configs.append(op_config.text)
+
+      wf_operation_config.append(op_configs)
+
+### Insert parsed information into the created tables
+  create_workflow_sql = f"""
+  INSERT INTO
+    {workflow_table_name} (id, state, template, title, description, parent,
+    creatorId, organizationId, dateCreated, dateCompleted, mediaPackage,
+    mediaPackageId, seriesId)
+  VALUES
+    ( %s, %s, %s, %s, %s, %s, %s, %s, %s, %s, %s, %s, %s )
+  """
+
+  create_workflow_configuration_sql = f"""
+  INSERT INTO
+    {workflow_configuration_table_name} (workflow_id, key_part, value_part)
+  VALUES
+    ( %s, %s, %s )
+  """
+
+  create_workflow_operation_sql = f"""
+  INSERT INTO
+    {workflow_operation_table_name} (id, template, job, state, description,
+    holdurl, holdActionTitle, failOnError, if_condition, unless_condition,
+    exceptionHandlerWorkflow, abortable, continuable, started, completed,
+    timeInQueue, maxAttempts, failedAttempts, executionHost, retryStrategy,
+    POSITION, INSTANCE_id, operations_ORDER )
+  VALUES
+    ( %s, %s, %s, %s, %s, %s, %s, %s, %s, %s,
+      %s, %s, %s, %s, %s, %s, %s, %s, %s, %s,
+      %s, %s, %s )
+  """
+
+  create_workflow_operation_configuration_sql = f"""
+  INSERT INTO
+    {workflow_operation_configuration_table_name} (workflow_operation_id, key_part, value_part)
+  VALUES
+    ( %s, %s, %s )
+  """
+
+  insert_parsed(create_workflow_sql, wf_items)
+  insert_parsed(create_workflow_configuration_sql, wf_config)
+  insert_parsed(create_workflow_operation_sql, wf_operation)
+  insert_parsed(create_workflow_operation_configuration_sql, wf_operation_config)
+  print(workflow_id)
+  wf_items = []
+  wf_config = []
+  wf_operation = []
+  wf_operation_config = []
+
+
+### Delete workflow information from oc_job
+print("Delete information from oc_job table...")
+### Get information from database
+select_id_from_job_table = """
+SELECT id FROM oc_job WHERE operation='START_WORKFLOW'
+"""
+ids = execute_read_query(connection, select_id_from_job_table)
+
+### Remove workflow XML from oc_job
+sql_update_job_payload_query = """
+UPDATE oc_job SET payload = %s where id = %s
+"""
+for id in ids:
+  execute_query_with_data(connection, sql_update_job_payload_query, (id, id))
```

Comment:
Why set `payload` to the entities primary key? If it's identical, we could just set it to `null`.